skyportal/handlers/api/public_pages/public_source_page.py

```diff
@@ -0,0 +1,267 @@
+import operator  # noqa: F401
+import json
+
+import joblib
+from sqlalchemy import or_
+
+from baselayer.app.access import auth_or_token, permissions
+from baselayer.log import make_log
+from ..source import get_source
+from ...base import BaseHandler
+
+from ....models import PublicSourcePage, Group, Stream, Classification, Photometry
+from ....utils.thumbnail import get_thumbnail_alt_link, get_thumbnail_header
+
+log = make_log('api/public_source_page')
+
+
+def calculate_hash(data):
+    return joblib.hash(json.dumps(data, sort_keys=True))
```

Comment:
Very nice that you thought about sorting the keys, definitely a requirements here!